src/main/java/com/thealgorithms/strings/AhoCorasick.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Aho-Corasick String Matching Algorithm Implementation
+ *
+ * This code implements the Aho-Corasick algorithm, which is used for efficient
+ * string matching in a given text. It can find multiple patterns simultaneously
+ * and records their positions in the text.
+ *
+ * Author: Prabhat-Kumar-42
+ * GitHub: https://github.com/Prabhat-Kumar-42
+ */
+
+package com.thealgorithms.strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+
+public class AhoCorasick {
+
+    // Trie Node Class
+    public class Node {

```

Comment:
```suggestion
    private static class Node {

```

Reply:
got it, will change it to private static.